lib/acts_as_taggable_on_mongoid/taggable/changeable.rb

```diff
@@ -60,11 +60,12 @@ def setters
       private
 
       def attribute_will_change!(attribute_name)
-        return super if tag_types.none? { |_tag_name, tag_definition| tag_definition.tag_list_name.to_s == attribute_name.to_s }
+        tag_definition = tag_types.detect { |_tag_name, tag_def| tag_def.tag_list_name.to_s == attribute_name.to_s }&.last
+        return super if tag_definition.blank?
 
         return if changed_attributes.key?(attribute_name)
 
-        changed_attributes[attribute_name] = public_send(attribute_name)&.dup
+        changed_attributes[attribute_name] = tag_list_cache_on(tag_definition)&.dup
```

Comment:
Seeing the cache here made me wonder about caching in general for tags. When this is integrated into MM, we should perhaps think about caching of tag lists and industry lists. As owner tag lists will not change often, we probably could probably save some perf here. Wrapping the methods in a rails cache?

Reply:
Out of scope for this PR.  This is something we'll have to keep in mind in MortgageMapp as we do the work there.